kale/embed/video_i3d.py

```diff
@@ -436,10 +480,29 @@ def extract_features(self, x):
         return self.avg_pool(x)
 
 
-def i3d(name, num_channels, pretrained=False, progress=True):
-    model = InceptionI3d(in_channels=num_channels)
+def i3d(name, num_channels, num_classes, pretrained=False, progress=True):
+    """Get InceptionI3d module w/o pretrained model."""
+    model = InceptionI3d(in_channels=num_channels, num_classes=num_classes)
 
     if pretrained:
         state_dict = load_state_dict_from_url(model_urls[name], progress=progress)
+        # delete logits.conv3d parameters due to different class number.
+        # state_dict.pop("logits.conv3d.weight")
+        # state_dict.pop("logits.conv3d.bias")
+        # model.load_state_dict(state_dict, strict=False)
         model.load_state_dict(state_dict)
     return model
+
+
+def i3d_joint(rgb_pt, flow_pt, num_classes, pretrained=False, progress=True):
+    """Get I3D models."""
+    i3d_rgb = i3d_flow = None
+    if rgb_pt is not None and flow_pt is None:
+        i3d_rgb = i3d(name=rgb_pt, num_channels=3, num_classes=num_classes, pretrained=pretrained, progress=progress)
+    elif rgb_pt is None and flow_pt is not None:
+        i3d_flow = i3d(name=flow_pt, num_channels=2, num_classes=num_classes, pretrained=pretrained, progress=progress)
+    elif rgb_pt is not None and flow_pt is not None:
```

Comment:
This is not necessary, right. Change the above if and elif two ifs will do.
if rgb_pt is not None: compute i3d_rgb
if flow_pt is not None: compute i3d_flow
Done.